upgrades/steps_28.go

```diff
@@ -64,3 +80,83 @@ func prepopulateRebootHandledFlagsForDeployedUnits(ctx Context) error {
 	}
 	return nil
 }
+
+func moveUniterStateToController(ctx Context) error {
```

Comment:
I was under the impression that you meant to do this as a fallback in the code that reads the uniter state. No objection to doing it as an upgrade step (guess it's cleaner) but I am curious as to why you decided to go with this approach.

Reply:
I had thought to do as a fall back but changed my mind after talking to @mitechie.  It's cleaner to do the upgrade step and lets us remove the old code as well.  Thought I'd give it a day to get the upgrade step working.  Looks like that's a go.